README.md
WordPress plugin that uses [exiftool](http://owl.phy.queensu.ca/~phil/exiftool/) to get the exif data of images files. 

Makes great use of: https://github.com/tsmgeek/ExifTool_PHP_Stayopen


Image-ExifTool-9.54/Changes
DO NOT EDIT THIS FILE -- it is generated from the html history files.

ExifTool Version History

RSS feed: http://owl.phy.queensu.ca/~phil/exiftool/rss.xml

Note: The most recent production release is Version 9.53. (Other versions are
considered development releases, and are not uploaded to CPAN.)

(I will be away March 7-15, so don't expect any email/forum responses
during this period, and the next release will likely be towards the end of
the month. - PH)

Mar. 1, 2014 - Version 9.54

  - Added a new Pentax LensType
  - Decode a few more FLIR tags
  - Generate missing default language tags for alternate-language QuickTime tags
    without them
  - Improved identification of some file sub-types based on filename extension
  - Fixed bug extracting alternate-language QuickTime tags with numerical ID's
    when processing multiple files in a single command
  - Fixed potential problem when using some of the recently added options with
    the -stay_open feature
  - Fixed problem loading a config file with a single quote in the file name
  - Fixed problem writing a shortcut tag when the target tag has a group name
  - Fixed problem recognizing HTML files with a leading UTF-8 BOM

Feb. 22, 2014 - Version 9.53 (production release)

  - Added user-defined file types feature
  - Added support for a few more XMP-expressionmedia tags
  - Decode CameraTemperature from a few new Canon models
  - Decode another Sony tag (thanks Jos Roost)
  - Recognize the JXR extension
  - Changed wording of IPTC "Unrecognized data" warning
  - Patched round-off error problem in seconds of some date/time values
  - Fixed problem creating EXV files with the -o option
  - Fixed problem conditionally replacing Extra block-type tags

Feb. 17, 2014 - Version 9.52

  - Fixed bug introduced in 9.44 which resulted in incorrect maker note offsets
    when copying maker notes from DNG images of some Pentax models

Feb. 15, 2014 - Version 9.51

  - Added a number of new CanonModelID values
  - Added a new Sony/Minolta LensType (thanks Jos Roost)
  - Added a new Panasonic BurstMode
  - Added a new Pentax LensType
  - Added new ColorSpaceTags shortcut
  - Decode a number of new PanasonicRaw tags (thanks LibRaw)
  - Decode ColorData tags for the new Canon EOS 1200D
  - Improved Sony maker note decoding for some newer models (thanks Jos Roost)
  - Fixed conversion of XMP:LensID for Pentax lenses on Ricoh-branded cameras

Feb. 8, 2014 - Version 9.50

  - Added support for Exiv2 EXV metadata files
  - Added ability to read/write/copy GeoTiff tags as a block
  - Added ability to combine -b with -j to write base64-encoded binary data in
    JSON output format
  - Added a new Nikon LensID
  - Added a new Sony/Minolta LensType (thanks Jos Roost)
  - Added new Extra ProcessingTime tag
  - Added a new Olympus CameraType
  - Added header signature to .EXIF files
  - Decode a new Olympus tag (thanks LibRaw)
  - Improved conversion of some Canon RecordMode values
  - Improved decoding of tags for Hasselblad Stellar (thanks Jos Roost)
  - Fixed -htmlDump offsets for information extracted from some embedded images
  - Fixed hemisphere problem in Composite GPS tags derived from
    QuickTime:LocationInformation
  - API Changes:
      - Option names are now case insensitive

Feb. 1, 2014 - Version 9.49

  - Added new Olympus and Samsung LensType values, and fixed an Olympus lens
    name (thanks Niels Kristian Bech Jensen)
  - Added another Polish tag translation (thanks Kacper Perschke)
  - Added a new Panasonic ColorEffect value
  - Decode a new FujiFilm tag (thanks LibRaw)
  - Adjusted scaling factor for H264 ExposureTime values (thanks Francois)

Jan. 25, 2014 - Version 9.48

  - Added a new GeoTiff tag
  - Added a couple of new Canon LensType values
  - Added a few new XMP DarwinCore tags
  - Updated Sony MakerNotes for the ILCE-5000 (thanks Jos Roost)
  - Improved Polish translations (thanks Kacper Perschke)
  - Fixed problem writing RicohSubdirIFD in GR images
  - Fixed runtime error when writing PNG image with a TIFF-format EXIF profile

Jan. 18, 2014 - Version 9.47

  - Added read support for JPEG-HDR APP11 information
  - Added read support for Media Jukebox APP9 information
  - Added a new Olympus LensType (thanks Niels Kristian Bech Jensen)
  - Added a few new Nikon LensID's (thanks Robert Rottmerhusen)
  - Added a new Minolta/Sony LensType and fixed some Tamron lens names (thanks
    Jos Roost)
  - Decode more FLIR tags (thanks Tomas)
  - Patched for quirk in Kodak PixPro AZ362 maker notes

Jan. 11, 2014 - Version 9.46 (production release)

  - Added a few Nikon lenses (thanks Niels for two)
  - Added a few new XMP-xmpDM tags (thanks Mats Peterson)
  - Added a new Pentax DriveMode (thanks Doug O'Brien)
  - Added a new Olympus lens (thanks Niels Kristian Bech Jensen)
  - Decode a number of new QuickTime tags
  - Decode a few new Canon filter tags
  - Improved -d option to properly handle time zones with %s and %z
  - Improved conversions for Panasonic Composite AdvancedSceneMode
  - Changed a few Tamron lens strings (thanks Niels Kristian Bech Jensen)
  - Patched to avoid round-off errors in seconds of filesystem time tags
  - Fixed extraction of CompressorID for some MOV videos

Dec. 21, 2013 - Version 9.45

  - Added details about movie data offset in MOV verbose dump
  - Added a new Pentax WhiteBalance value
  - Added a new Sony/Minolta LensType (thanks Jos Roost)
  - Decode a new QuickTime tag
  - Prevent illegal tag names to be generated from user-defined XMP tags
  - Patched to allow reading some improperly formatted EXIF UserComment values
  - Fixed problem where reading some large M2TS files could take a loooong time

Dec. 11, 2013 - Version 9.44

  - Added a new Pentax WhiteBalance value
  - Added back the K-3 SRResult tag, but make it Unknown and with no print
    conversion
  - Fixed problem copying the maker notes of some recent Pentax models

Dec. 8, 2013 - Version 9.43

  - Added a new Olympus CameraType

Image-ExifTool-9.54/lib/Image/ExifTool/Shortcuts.pm
#------------------------------------------------------------------------------
# File:         Shortcuts.pm
#
# Description:  ExifTool shortcut tags
#
# Revisions:    02/07/2004 - PH Moved out of Exif.pm
#               09/15/2004 - PH Added D70Boring from Greg Troxel
#               01/11/2005 - PH Added Canon20D from Christian Koller
#               03/03/2005 - PH Added user defined shortcuts
#               03/26/2005 - PH Added Nikon from Tom Christiansen
#               02/28/2007 - PH Removed model-dependent shortcuts
#                            --> this is what UserDefined::Shortcuts is for
#               02/25/2009 - PH Added Unsafe
#               07/03/2010 - PH Added CommonIFD0
#------------------------------------------------------------------------------

package Image::ExifTool::Shortcuts;

use strict;
use vars qw($VERSION);

$VERSION = '1.50';

# this is a special table used to define command-line shortcuts
# (documentation Notes may be added for these via %shortcutNotes in BuildTagLookup.pm)
%Image::ExifTool::Shortcuts::Main = (
    # this shortcut allows the three common date/time tags to be shifted at once
    AllDates => [
        'DateTimeOriginal',
        'CreateDate',
        'ModifyDate',
    ],
    # This is a shortcut to some common information which is useful in most images
    Common => [
        'FileName',
        'FileSize',
        'Model',
        'DateTimeOriginal',
        'ImageSize',
        'Quality',
        'FocalLength',
        'ShutterSpeed',
        'Aperture',
        'ISO',
        'WhiteBalance',
        'Flash',
    ],
    # This shortcut provides the same information as the Canon utilities
    Canon => [
        'FileName',
        'Model',
        'DateTimeOriginal',
        'ShootingMode',
        'ShutterSpeed',
        'Aperture',
        'MeteringMode',
        'ExposureCompensation',
        'ISO',
        'Lens',
        'FocalLength',
        'ImageSize',
        'Quality',
        'Flash',
        'FlashType',
        'ConditionalFEC',
        'RedEyeReduction',
        'ShutterCurtainHack',
        'WhiteBalance',
        'FocusMode',
        'Contrast',
        'Sharpness',
        'Saturation',
        'ColorTone',
        'ColorSpace',
        'LongExposureNoiseReduction',
        'FileSize',
        'FileNumber',
        'DriveMode',
        'OwnerName',
        'SerialNumber',
    ],
    Nikon => [
        'Model',
        'SubSecDateTimeOriginal',
        'ShutterCount',
        'LensSpec',
        'FocalLength',
        'ImageSize',
        'ShutterSpeed',
        'Aperture',
        'ISO',
        'NoiseReduction',
        'ExposureProgram',
        'ExposureCompensation',
        'WhiteBalance',
        'WhiteBalanceFineTune',
        'ShootingMode',
        'Quality',
        'MeteringMode',
        'FocusMode',
        'ImageOptimization',
        'ToneComp',
        'ColorHue',
        'ColorSpace',
        'HueAdjustment',
        'Saturation',
        'Sharpness',
        'Flash',
        'FlashMode',
        'FlashExposureComp',
    ],
    # This shortcut may be useful when copying tags between files to either
    # copy the maker notes as a block or prevent it from being copied
    MakerNotes => [
        'MakerNotes',   # (for RIFF MakerNotes)
        'MakerNoteApple',
        'MakerNoteCanon',
        'MakerNoteCasio',
        'MakerNoteCasio2',
        'MakerNoteFLIR',
        'MakerNoteFujiFilm',
        'MakerNoteGE',
        'MakerNoteGE2',
        'MakerNoteHasselblad',
        'MakerNoteHP',
        'MakerNoteHP2',
        'MakerNoteHP4',
        'MakerNoteHP6',
        'MakerNoteISL',
        'MakerNoteJVC',
        'MakerNoteJVCText',
        'MakerNoteKodak1a',
        'MakerNoteKodak1b',
        'MakerNoteKodak2',
        'MakerNoteKodak3',
        'MakerNoteKodak4',
        'MakerNoteKodak5',
        'MakerNoteKodak6a',
        'MakerNoteKodak6b',
        'MakerNoteKodak7',
        'MakerNoteKodak8a',
        'MakerNoteKodak8b',
        'MakerNoteKodak8c',
        'MakerNoteKodak9',
        'MakerNoteKodak10',
        'MakerNoteKodakUnknown',
        'MakerNoteKyocera',
        'MakerNoteMinolta',
        'MakerNoteMinolta2',
        'MakerNoteMinolta3',
        'MakerNoteNikon',
        'MakerNoteNikon2',
        'MakerNoteNikon3',
        'MakerNoteOlympus',
        'MakerNoteOlympus2',
        'MakerNoteLeica',
        'MakerNoteLeica2',
        'MakerNoteLeica3',
        'MakerNoteLeica4',
        'MakerNoteLeica5',
        'MakerNoteLeica6',
        'MakerNotePanasonic',
        'MakerNotePanasonic2',
        'MakerNotePentax',
        'MakerNotePentax2',
        'MakerNotePentax3',
        'MakerNotePentax4',
        'MakerNotePentax5',
        'MakerNotePentax6',
        'MakerNotePhaseOne',
        'MakerNoteReconyx',
        'MakerNoteRicoh',
        'MakerNoteRicohText',
        'MakerNoteSamsung1a',
        'MakerNoteSamsung1b',
        'MakerNoteSamsung2',
        'MakerNoteSanyo',
        'MakerNoteSanyoC4',
        'MakerNoteSanyoPatch',
        'MakerNoteSigma',
        'MakerNoteSony',
        'MakerNoteSony2',
        'MakerNoteSony3',
        'MakerNoteSony4',
        'MakerNoteSony5',
        'MakerNoteSonyEricsson',
        'MakerNoteSonySRF',
        'MakerNoteUnknownText',
        'MakerNoteUnknownBinary',
        'MakerNoteUnknown',
    ],
    # "unsafe" tags we normally don't copy in JPEG images, defined
    # as a shortcut to use when rebuilding JPEG EXIF from scratch
    Unsafe => [
        'IFD0:YCbCrPositioning',
        'IFD0:YCbCrCoefficients',
        'IFD0:TransferFunction',
        'ExifIFD:ComponentsConfiguration',
        'ExifIFD:CompressedBitsPerPixel',
        'InteropIFD:InteropIndex',
        'InteropIFD:InteropVersion',
        'InteropIFD:RelatedImageWidth',
        'InteropIFD:RelatedImageHeight',
    ],
    # standard tags used to define the color space of an image
    # (useful to preserve color space when deleting all meta information)
    ColorSpaceTags => [
        'ExifIFD:ColorSpace',
        'ExifIFD:Gamma',
        'InteropIFD:InteropIndex',
        'ICC_Profile',
    ],
    # common metadata tags found in IFD0 of TIFF images
    CommonIFD0 => [
        # standard EXIF
        'IFD0:ImageDescription',
        'IFD0:Make',
        'IFD0:Model',
        'IFD0:Software',
        'IFD0:ModifyDate',
        'IFD0:Artist',
        'IFD0:Copyright',
        # other TIFF tags
        'IFD0:Rating', 
        'IFD0:RatingPercent',
        'IFD0:DNGLensInfo',
        'IFD0:PanasonicTitle',
        'IFD0:PanasonicTitle2',
        'IFD0:XPTitle',
        'IFD0:XPComment',
        'IFD0:XPAuthor',
        'IFD0:XPKeywords',
        'IFD0:XPSubject',
    ],
);

#------------------------------------------------------------------------------
# load user-defined shortcuts if available
# Inputs: reference to user-defined shortcut hash
sub LoadShortcuts($)
{
    my $shortcuts = shift;
    my $shortcut;
    foreach $shortcut (keys %$shortcuts) {
        my $val = $$shortcuts{$shortcut};
        # also allow simple aliases
        $val = [ $val ] unless ref $val eq 'ARRAY';
        # save the user-defined shortcut or alias
        $Image::ExifTool::Shortcuts::Main{$shortcut} = $val;
    }
}
# (for backward compatibility, renamed in ExifTool 7.75)
if (%Image::ExifTool::Shortcuts::UserDefined) {
    LoadShortcuts(\%Image::ExifTool::Shortcuts::UserDefined);
}
if (%Image::ExifTool::UserDefined::Shortcuts) {
    LoadShortcuts(\%Image::ExifTool::UserDefined::Shortcuts);
}


1; # end

__END__

=head1 NAME

Image::ExifTool::Shortcuts - ExifTool shortcut tags

=head1 SYNOPSIS

This module is required by Image::ExifTool.

=head1 DESCRIPTION

This module contains definitions for tag name shortcuts used by
Image::ExifTool.  You can customize this file to add your own shortcuts.

Individual users may also add their own shortcuts to the .ExifTool_config
file in their home directory (or the directory specified by the
EXIFTOOL_HOME environment variable).  The shortcuts are defined in a hash
called %Image::ExifTool::UserDefined::Shortcuts.  The keys of the hash are
the shortcut names, and the elements are either tag names or references to
lists of tag names.

An example shortcut definition in .ExifTool_config:

    %Image::ExifTool::UserDefined::Shortcuts = (
        MyShortcut => ['createdate','exif:exposuretime','aperture'],
        MyAlias => 'FocalLengthIn35mmFormat',
    );

In this example, MyShortcut is a shortcut for the CreateDate,
EXIF:ExposureTime and Aperture tags, and MyAlias is a shortcut for
FocalLengthIn35mmFormat.

The target tag names may contain an optional group name prefix.  A group
name applied to the shortcut will be ignored for any target tag with a group
name prefix.

=head1 AUTHOR

Copyright 2003-2014, Phil Harvey (phil at owl.phy.queensu.ca)

This library is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

=head1 SEE ALSO

L<Image::ExifTool(3pm)|Image::ExifTool>

=cut
